### ISSUE TITLE
Hard coded "A" query?

### ISSUE THREAD

**Issue by lojikil** [st3fan/tinydoh#3]
https://github.com/st3fan/tinydoh/blob/0b51cf6c47466691dc76a6ef8fc023e117101f41/main.go#L96

I apologize if this is just me misunderstand the DoH draft, but is a protocol restriction to have the "A" query hardcoded here?

(I'm happy to figure it out, implement, and PR, I just wanted to make sure that I'm not misunderstanding something. Reading Protocol documents before coffee at 0500 local is not a good look for me...)

Thanks!

**Comment by st3fan** [comment]
No, I actually think this is incorrect. The specification does not say anything about specific queries, it just says to forward the query as-is. This hard coded query works for Firefox because it only requests A (and AAAA).

We should probably revisit that code.

**Comment by lojikil** [comment]
I _think_ it can follow the same pattern as the `POST` section, no? 

1. Base64 unencode the parameter
1. decode the DNS message
1. split out `t` and `n` from the message

I can try that & issue a PR if that works, fair?
